src/code/z_draw.c

```diff
@@ -1,49 +1,842 @@
+#include "assets/objects/object_gi_reserve_c_00/object_gi_reserve_c_00.h"
+#include "assets/objects/object_gi_reserve_c_01/object_gi_reserve_c_01.h"
+#include "assets/objects/object_gi_reserve00/object_gi_reserve00.h"
+#include "assets/objects/object_gi_reserve01/object_gi_reserve01.h"
+#include "assets/objects/object_gi_rupy/object_gi_rupy.h"
+#include "assets/objects/object_gi_schedule/object_gi_schedule.h"
+#include "assets/objects/object_gi_seahorse/object_gi_seahorse.h"
+#include "assets/objects/object_gi_shield_2/object_gi_shield_2.h"
+#include "assets/objects/object_gi_shield_3/object_gi_shield_3.h"
+#include "assets/objects/object_gi_soldout/object_gi_soldout.h"
+#include "assets/objects/object_gi_soul/object_gi_soul.h"
+#include "assets/objects/object_gi_stick/object_gi_stick.h"
+#include "assets/objects/object_gi_stonemask/object_gi_stonemask.h"
+#include "assets/objects/object_gi_sutaru/object_gi_sutaru.h"
+#include "assets/objects/object_gi_sword_1/object_gi_sword_1.h"
+#include "assets/objects/object_gi_sword_2/object_gi_sword_2.h"
+#include "assets/objects/object_gi_sword_3/object_gi_sword_3.h"
+#include "assets/objects/object_gi_sword_4/object_gi_sword_4.h"
+#include "assets/objects/object_gi_truth_mask/object_gi_truth_mask.h"
+#include "assets/objects/object_gi_zoramask/object_gi_zoramask.h"
+#include "assets/objects/object_bsmask/object_bsmask.h"
+#include "assets/objects/object_st/object_st.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_draw/GetItem_Draw.s")
+void GetItem_DrawBombchu(PlayState* play, s16 drawId);
+void GetItem_DrawPoes(PlayState* play, s16 drawId);
+void GetItem_DrawFairy2(PlayState* play, s16 drawId);
+void GetItem_DrawSkullToken(PlayState* play, s16 drawId);
+void GetItem_DrawCompass(PlayState* play, s16 drawId);
+void GetItem_DrawPotion(PlayState* play, s16 drawId);
+void GetItem_DrawGoronSword(PlayState* play, s16 drawId);
+void GetItem_DrawDekuNuts(PlayState* play, s16 drawId);
+void GetItem_DrawRecoveryHeart(PlayState* play, s16 drawId);
+void GetItem_DrawFish(PlayState* play, s16 drawId);
+void GetItem_DrawOpa0(PlayState* play, s16 drawId);
+void GetItem_DrawOpa0Xlu1(PlayState* play, s16 drawId);
+void GetItem_DrawOpa01(PlayState* play, s16 drawId);
+void GetItem_DrawXlu01(PlayState* play, s16 drawId);
+void GetItem_DrawSeaHorseCaught(PlayState* play, s16 drawId);
+void GetItem_DrawFairy(PlayState* play, s16 drawId);
+void GetItem_DrawMoonsTear(PlayState* play, s16 drawId);
+void GetItem_DrawMagicArrow(PlayState* play, s16 drawId);
+void GetItem_DrawUpgrades(PlayState* play, s16 drawId);
+void GetItem_DrawRupee(PlayState* play, s16 drawId);
+void GetItem_DrawSmallRupee(PlayState* play, s16 drawId);
+void GetItem_DrawWallet(PlayState* play, s16 drawId);
+void GetItem_DrawRemains(PlayState* play, s16 drawId);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_draw/func_800EE364.s")
+typedef struct {
+    /* 0x00 */ void (*drawFunc)(PlayState*, s16);
+    /* 0x04 */ Gfx* dLists[8];
```

Comment:
Agreed with Elliptic. This should be able to be `void*` without too much difficulty. Might want to think about renaming it something else as well since it is not just `DL`s, but nothing great comes to mind at the moment.

Reply:
Agreed.
Maybe call it something like `params` or similar?

Reply:
What about `dListsOrMtx`?

Reply:
That just seems strange to me. An idea I had was maybe `drawArgs` since it is like the arguments passed into the functions used?

Reply:
I think argument-based naming's a bit too far from its usual concrete meaning. I think I want something like `drawResourses`, but not actually "resources".

Reply:
Building on that, maybe just `drawAssets`?

Reply:
Maybe, although I wouldn't ever call an Mtx an asset in our usual sense.